### DISCUSSION TITLE
image NBD serving

### DISCUSSION THREAD

**Discussion by SimStone** [ltsp/ltsp#743] [Q&A] [answered]
Hi all,
Once created, is it possible to serve the image using NBD (like in LTSP5) rather than NFS ?

**Comment by alkisg** [comment, marked as answer]
Sure, it's just not automated yet. You'd need:

1. To install the nbd-server package in the LTSP server,
2. the /etc/nbd-server/conf.d directory from an LTSP5 installation,
3. to install the nbd-client package in the LTSP image/chroot,
4. and in /srv/tftp/ltsp/ltsp.ipxe, to change the nfsroot= kernel parameter to nbdroot=.

**Reply by SimStone** [reply]
Thanks a lot,

**Comment by ElBarbossa** [comment]
Hello,

We have set up an ltsp6 ubuntu 2204 server with an external dhcp.
I am looking to start clients in nbd service.
we already have a ltsp5 ubuntu 1804 server in production with nbd-server.

However when I reproduce the same configuration with ubuntu 2204 ltsp6 I get a /dev/nbd0 with the busybox.

I submit to you the configuration of our configuration pxelinux.cfg/default

We have followed your solution but it remains blocked at busybox.

Thank you

```
LABEL test
        MENU DEFAULT
        MENU LABEL TEST
        TEXT HELP
                Boots /srv/ltsp/images/assistance.img via NBD.
        ENDTEXT
        MENU INDENT 2
        KERNEL assistance/vmlinuz
        APPEND root=/dev/nbd0 nbdroot=/srv/ltsp/assistance ltsp.image=images/assistance.img noapic diskless fastboot ip=dhcp loop.max_part=9
        INITRD ltsp.img,assistance/initrd.img
        IPAPPEND 3


```

**Comment by alkisg** [comment]
The syntax for the nbdroot= parameter isn't the same as the nfsroot= parameter; put the same that ltsp5 used there.

Also there's no reason to use pxelinux, you may keep using ipxe.
